### DISCUSSION TITLE
public form allowing only one response per recipient?

### DISCUSSION THREAD

**Discussion by nathanielconroy** [Budibase/budibase#5943] [Help]
Is it possible to create a public form in budibase that can only be used once per recipient? Other applications usually accomplish this by generating a different URL containing a "secret" for each recipient and then only allowing one response per URL. Does something like this exist in budibase, or is there an alternative way to accomplish something similar? 

I'd prefer not to force each user to authenticate before submitting a response, since most of the recipients will not have dedicated accounts.

**Comment by melohagan** [comment]
Hi @nathanielconroy 

There's no specific feature for this, but it could be built. Some starting points to look at:

- https://docs.budibase.com/docs/url-parameters (to allow you to pass a unique form reference)
- https://docs.budibase.com/docs/email (to send out the form links)
- https://docs.budibase.com/docs/conditions (show the form if not completed, show a 'completed' message otherwise)

You would also need to create a table to keep track of:
- unique form reference
- isComplete boolean

**Reply by nathanielconroy** [reply]
Thanks! I'll see if I can mock something up using these features.

**Reply by melohagan** [reply]
Awesome, I'd love to see what you come up with in the end!
If you get stuck at any point along the way don't hesitate to reach out. :)
